mmocr/models/textrecog/backbones/resnet.py

```diff
@@ -0,0 +1,232 @@
+            self.res_layers.append(layer_name)
+
+    def _make_layer(self, block_cfgs, inplanes, planes, blocks, stride):
+        layers = []
+        downsample = None
+        block_cfgs_ = block_cfgs.copy()
+        if isinstance(stride, int):
+            stride = (stride, stride)
+
+        if stride[0] != 1 or stride[1] != 1 or inplanes != planes:
+            downsample = ConvModule(
+                inplanes,
+                planes,
+                1,
+                stride,
+                norm_cfg=dict(type='BN'),
+                act_cfg=None)
+
+        if block_cfgs_['type'] == 'BasicBlock':
+            block = BasicBlock
+            block_cfgs_.pop('type')
+        else:
+            raise ValueError('{} not implement yet'.format(block['type']))
+
+        layers.append(
+            block(
+                inplanes,
+                planes,
+                stride=stride,
+                downsample=downsample,
+                **block_cfgs_))
+        inplanes = planes
+        for _ in range(1, blocks):
+            layers.append(block(inplanes, planes, **block_cfgs_))
+
+        return Sequential(*layers)
+
+    def _make_stem_layer(self, in_channels, stem_channels):
+        if isinstance(stem_channels, int):
+            stem_channels = [stem_channels]
+        stem_layers = []
+        for _, channels in enumerate(stem_channels):
+            stem_layer = ConvModule(
+                in_channels,
+                channels,
+                kernel_size=3,
+                stride=1,
+                padding=1,
+                bias=False,
+                norm_cfg=dict(type='BN'),
+                act_cfg=dict(type='ReLU'))
+            in_channels = channels
+            stem_layers.append(stem_layer)
+        self.stem_layers = Sequential(*stem_layers)
+        self.inplanes = stem_channels[-1]
+
+    def _make_stage_plugins(self, plugins, stage_idx):
+        """Make plugins for ResNet ``stage_idx`` th stage.
+
+        Currently we support to insert ``nn.Maxpooling``,
+        ``nn.conv2d``into the backbone. This is designed for ResNet31
+        architecture likes
+
+        An example of plugins format could be:
+
+        Examples:
+            >>> plugins=[
+            ...     dict(cfg=dict(type="Maxpooling", arg=(2,2)),
+            ...          stages=(True, True, False, False),
+            ...          position='before_stage'),
+            ...     dict(cfg=dict(type="Maxpooling", arg=(2,1)),
+            ...          stages=(False, False, True, Flase),
+            ...          position='before_stage'),
+            ...     dict(cfg=dict(type="Conv", args=(3,1,1)),
+            ...          stages=(True, True, True, True),
+            ...          position=('after_stage')
+            ... ]
+
+        Suppose ``stage_idx=1``, the structure of stage would be:
+
+        .. code-block:: none
+
+            Maxpooling-> Basicblocks * blocks[stage_idx - 1] -> Conv
```

Comment:
```suggestion
            Maxpooling -> Basicblocks * blocks[stage_idx - 1] -> Conv
```
Where is the second maxpooling layer?

Reply:
The second maxpooling layer is only used at stage3 in ResNet31

Reply:
The first maxpooling layer's postion is  (True, True, False, False) and the second one is (False, False, True, False). So there won't be two maxpooling layers at the same stage